cmds/cpu/cpu9p.go

```diff
@@ -144,8 +145,21 @@ func (l *cpu9p) ReadAt(p []byte, offset int64) (int, error) {
 }
 
 // Write implements p9.File.WriteAt.
+// There is a very rare case where O_APPEND files are written more than
```

Comment:
what's interesting is Go's WriteAt fails anytime O_APPEND was set, yet we're able to get a single write through.

my guess is that Create wasn't passing the O_APPEND along, such that if you created a file with O_APPEND, the Go runtime wouldn't have f.appendmode set, so WriteAt would work. 

from looking at os file.go, there's a Create that makes up its own O_flags, and O_APPEND isn't one of them.  so i guess if you wanted to do O_CREATE | O_APPEND, you'd have to use OpenFile or whatever.  arguably, no one really wants to do that, and the only reason the shell does it is that it's either CREATE or APPEND.  

though there is a subtle difference between a freshly created file without append, and a freshly created file with append: for the latter, the OS will automatically seek your file for you to the end of the inode.  so if you have two threads writing to a log file, you'd get the corruption that presumably Go is trying to avoid.  i.e. you'd have to do OpenFile(O_CREATE | O_APPEND) so that the creator is also in append mode.

anyway, this is subtle shit, and mildly annoying that you have to work around the runtime.  imagine if glibc wouldn't let you do a pwrite() to an O_APPEND fd!

in go's defense, posix and linux disagree on the behavior of pwrite and O_APPEND:
```
BUGS
       POSIX requires that opening a file with the O_APPEND flag should have no effect on the location at which pwrite() writes data.  However, on Linux, if  a
       file is opened with O_APPEND, pwrite() appends data to the end of the file, regardless of the value of offset.

```